File: samples/Workbench/SampleData.cs
using Microsoft.Data.Sqlite;

namespace Workbench;

/// <summary>Creates deterministic sample data so every workbench checkout demonstrates the same reports.</summary>
public static class SampleData
{
    private static readonly string[] Customers =
    [
        "Acme Corp", "Globex", "Initech", "Umbrella Group", "Stark Industries", "Wayne Enterprises",
        "Tyrell Corp", "Wonka Industries", "Cyberdyne Systems", "Aperture Science", "Soylent Corp", "Hooli",
    ];

    private static readonly string[] Regions = ["NORTH", "SOUTH", "EAST", "WEST"];
    private static readonly string[] Statuses = ["NEW", "PENDING", "SHIPPED", "CANCELLED"];

    /// <summary>
    /// Creates the sample SQLite database and inserts 500 deterministic orders when the table is empty.
    /// </summary>
    /// <param name="dbPath">The SQLite database-file path to initialize.</param>
    /// <remarks>Creates the parent directory and database file when absent. Existing non-empty order data is left unchanged.</remarks>
    public static void EnsureSeeded(string dbPath)
    {
        Directory.CreateDirectory(Path.GetDirectoryName(dbPath)!);

        using var conn = new SqliteConnection($"Data Source={dbPath}");
        conn.Open();

        using (var create = conn.CreateCommand())
        {
            create.CommandText = """
                CREATE TABLE IF NOT EXISTS ORDERS (
                    ORDER_ID   INTEGER PRIMARY KEY,
                    CUSTOMER   TEXT    NOT NULL,
                    REGION     TEXT    NOT NULL,
                    STATUS     TEXT    NOT NULL,
                    AMOUNT     NUMERIC NOT NULL,
                    ORDER_DATE TEXT    NOT NULL,
                    NOTES      TEXT    NULL
                );
                """;
            create.ExecuteNonQuery();
        }

        using (var count = conn.CreateCommand())
        {
            count.CommandText = "SELECT COUNT(*) FROM ORDERS";
            if (Convert.ToInt64(count.ExecuteScalar()) > 0) return;
        }

        var rng = new Random(42);
        var start = new DateOnly(2025, 1, 1);

        using var tx = conn.BeginTransaction();
        using var insert = conn.CreateCommand();
        insert.Transaction = tx;
        insert.CommandText = """
            INSERT INTO ORDERS (CUSTOMER, REGION, STATUS, AMOUNT, ORDER_DATE, NOTES)
            VALUES (@customer, @region, @status, @amount, @date, @notes)
            """;
        var pCustomer = insert.Parameters.Add("@customer", SqliteType.Text);
        var pRegion = insert.Parameters.Add("@region", SqliteType.Text);
        var pStatus = insert.Parameters.Add("@status", SqliteType.Text);
        var pAmount = insert.Parameters.Add("@amount", SqliteType.Real);
        var pDate = insert.Parameters.Add("@date", SqliteType.Text);
        var pNotes = insert.Parameters.Add("@notes", SqliteType.Text);

        for (var i = 0; i < 500; i++)
        {
            pCustomer.Value = Customers[rng.Next(Customers.Length)];
            pRegion.Value = Regions[rng.Next(Regions.Length)];
            pStatus.Value = Statuses[rng.Next(Statuses.Length)];
            pAmount.Value = Math.Round(rng.NextDouble() * 24_990 + 10, 2);
            pDate.Value = start.AddDays(rng.Next(550)).ToString("yyyy-MM-dd");
            pNotes.Value = rng.Next(4) == 0 ? (object)$"Priority handling #{i}" : DBNull.Value;
            insert.ExecuteNonQuery();
        }

        tx.Commit();
    }
}

File: src/demo/sample-data.js
// Deterministic sample data generator matching samples/Workbench/SampleData.cs.

const Customers = [
    "Acme Corp", "Globex", "Initech", "Umbrella Group", "Stark Industries", "Wayne Enterprises",
    "Tyrell Corp", "Wonka Industries", "Cyberdyne Systems", "Aperture Science", "Soylent Corp", "Hooli",
];

const Regions = ["NORTH", "SOUTH", "EAST", "WEST"];
const Statuses = ["NEW", "PENDING", "SHIPPED", "CANCELLED"];

/**
 * Simple pseudo-random number generator with fixed seed (matches Random(42) in C#).
 */
class LcgRandom {
    constructor(seed = 42) {
        this.state = seed % 2147483647;
        if (this.state <= 0) this.state += 2147483646;
    }

    next() {
        this.state = (this.state * 16807) % 2147483647;
        return (this.state - 1) / 2147483646;
    }

    nextInt(max) {
        return Math.floor(this.next() * max);
    }
}

/**
 * Creates the ORDERS table and seeds 500 deterministic orders into the SQLite database.
 *
 * @param {object} db
 */
export function seedSampleOrders(db) {
    db.run(`
        CREATE TABLE IF NOT EXISTS ORDERS (
            ORDER_ID   INTEGER PRIMARY KEY AUTOINCREMENT,
            CUSTOMER   TEXT    NOT NULL,
            REGION     TEXT    NOT NULL,
            STATUS     TEXT    NOT NULL,
            AMOUNT     NUMERIC NOT NULL,
            ORDER_DATE TEXT    NOT NULL,
            NOTES      TEXT    NULL
        );
    `);

    const count = db.queryScalar("SELECT COUNT(*) FROM ORDERS");
    if (count > 0) return;

    const rng = new LcgRandom(42);
    const startDate = new Date(Date.UTC(2025, 0, 1));

    for (let i = 0; i < 500; i++) {
        const customer = Customers[rng.nextInt(Customers.length)];
        const region = Regions[rng.nextInt(Regions.length)];
        const status = Statuses[rng.nextInt(Statuses.length)];
        const amount = Math.round((rng.next() * 24990 + 10) * 100) / 100;

        const dayOffset = rng.nextInt(550);
        const orderDate = new Date(startDate.getTime() + dayOffset * 86400000)
            .toISOString()
            .slice(0, 10);

        const notes = rng.nextInt(4) === 0 ? `Priority handling #${i}` : null;

        db.run(
            `INSERT INTO ORDERS (CUSTOMER, REGION, STATUS, AMOUNT, ORDER_DATE, NOTES) VALUES (?, ?, ?, ?, ?, ?)`,
            [customer, region, status, amount, orderDate, notes]
        );
    }
}
